File: 1-TwoSum/1-TwoSum.py
# Last updated: 7/14/2026, 1:55:06 PM
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen_nums={}
        for i,num in enumerate(nums):
            complement = target-num
            if complement in seen_nums:
                return [seen_nums[complement],i]
            else:
                seen_nums[num]=i

File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/14/2026, 1:55:07 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int complement=target-nums[i];
            if(numMap.containsKey(complement)){
                return new int[]{numMap.get(complement),i};
            }
            numMap.put(nums[i],i);
        }
        return new int[]{};
    }
}